mypy/treetransform.py

```diff
@@ -113,7 +127,13 @@ def visit_func_def(self, node: FuncDef) -> FuncDef:
         new.is_class = node.is_class
         new.is_property = node.is_property
         new.original_def = node.original_def
-        return new
+
+        if node in self.func_map:
+            result = self.func_map[node]
+            result.__dict__ = new.__dict__
```

Comment:
This is worth explaining, as this is a little unusual. In particular, this only works if the classes of the instances are identical (no subclasses)?